distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Period.enso

```diff
@@ -0,0 +1,176 @@
+         java.time.Period.
+       - start_date: Starting Date
+    Period_With_Date internal_period start_date
+
+    ## Get the portion of the period expressed in years.
+    years : Integer
+    years self = self.internal_period.getYears
+
+    ## Get the portion of the period expressed in months.
+    months : Integer
+    months self = self.internal_period.getMonths
+
+    ## Get the portion of the period expressed in days.
+    days : Integer
+    days self = self.internal_period.getDays
+
+    ## Add the specified amount of time to this period.
+
+       Arguments:
+       - other_period: The period to add to `self`. Note that this cannot be a
+         `Duration`, neither `Date_Time`.
+
+       > Example
+       Add 1 day to 1 month.
+
+           import Standard.Base.Data.Time.Period
+
+           example_add = (Period.months 1) + (Period.days 1)
+    + : Period -> Period ! (Time_Error | Illegal_Argument_Error)
+    + self other_period =
+        ensure_period other_period <|
+            Panic.catch Any (Period.Period_Without_Date (self.internal_period.plus other_period.internal_period)) err->
+                case err of
+                    _ : DateTimeException -> Error.throw Time_Error_Data "Period addition failed:"+err.getMessage
+                    _ : ArithmeticException -> Error.throw Illegal_Argument_Error_Data "Arithmetic error:"+err.getMessage cause=err
+
+    ## Subtract a specified amount of time from this period.
+
+       Arguments:
+       - other_period: Other Period to add to this Period. Note that this
+         cannot be a `Duration`, neither `Date_Time`.
+
+       > Example
+         Subtract 11 months from a period of 3 years
+
+             import Standard.Base.Data.Time.Period
+
+             example_subtract = (Period.years 3) - (Period.months 11)
+
+    - : Period -> Period ! (Time_Error | Illegal_Argument_Error)
+    - self other_period =
+        ensure_period other_period <|
+            Panic.catch Any (Period.Period_Without_Date (self.internal_period.minus other_period.internal_period)) err->
+                case err of
+                    DateTimeException -> Error.throw Time_Error_Data "Period subtraction failed"
+                    ArithmeticException -> Error.throw Illegal_Argument_Error_Data "Arithmetic error"
+
+    ## Check two periods for equality.
+
+       Note that two periods are equal if they have the exact same amount of
+       years, months, and days. So `(Period.days 30)` and
+       `(Period.new months=1`) are not equal.
+
+       Arguments:
+       - other_period: The period to compare against `self`.
+    == : Period -> Boolean ! (Time_Error | Illegal_Argument_Error)
+    == self that =
+        ensure_period that <|
+            (self.compare_to that) == Ordering.Equal
+
+    ## Compare this period to another period to produce an ordering.
+    compare_to : Period -> Ordering ! (Time_Error | Illegal_Argument_Error)
+    compare_to self that =
```

Comment:
Really not sure about this approach.

I feel a period should be detached from starting date. If a user wants to do that, they can do:
`start_date.add period_a . compare_to (start_date.add period_b)`.

If we want to have `compare_to` I suggest:
- If both have `self.days==0` then `self.total_months.compare_to other.total_months`.
- If both have `self.total_months==0` then `self.days.compare_to other.days`.
- Otherwise throw and `Incomparable_Values_Error`.

Reply:
If you want to detach period from starting date, I respect that. However, in that case, I would suggest that we don't implement `compare_to` at all (based on my reasoning from https://github.com/enso-org/enso/pull/3818/files/2e516dac89b74854a3b9e7cf44c982da96f4555d#r1006778870 and https://github.com/enso-org/enso/pull/3818#discussion_r1004444974), or that `compare_to` would just blindly throw an error.

The whole idea of attaching starting date to a period in some situations was a convenience for the users, so they don't have to implement the comparison of two periods from a certain start_date themselves.

cc @radeusgd @hubertp

Reply:
I'm happy to not implement `compare_to` I think better to make users be explicit of what they want the comparison to be,